### PR TITLE
112 render receipts tables

### PR DESCRIPTION
**Pull request by NemTechCN**

### CODE REVIEW

File: src/service/receipt.service.ts [hunk 1]
```diff
@@ -29,8 +32,24 @@ import {
 } from 'nem2-sdk';
 
 export class ReceiptService {
+    private transactionStatementsTable: HorizontalTable;
```
Comment (dgarcia360) [inline src/service/receipt.service.ts]: I'd create one class per transaction table, so we can test each table in the future.

Example: https://github.com/nemtech/nem2-cli/blob/master/src/commands/block/header.ts#L36

File: src/service/receipt.service.ts [hunk 2]
```diff
@@ -41,41 +60,53 @@ export class ReceiptService {
...
         statement.transactionStatements.map((transaction: TransactionStatement, transactionIndex: number) => {
-            txt += 'height:\t\t' + transaction.height + '\n';
             transaction.receipts.map((receipt: any, receiptIndex: number) => {
-                txt += '<index: ' + transactionIndex + '-' + receiptIndex + '>\t';
                 if (receipt instanceof BalanceTransferReceipt) {
-                    txt += 'version:\t' + receipt.version + '\n';
-                    txt += '\t\ttype:\t\t' + ReceiptType[receipt.type]  + '\n';
-                    txt += '\t\trecipientAddress:\t' +
+                    this.transactionStatementsTable.push([
+                        receipt.version as Table.CellOptions,
```
Comment (dgarcia360) [inline src/service/receipt.service.ts]: What does as Table.CellOptions mean?

File: src/service/receipt.service.ts [hunk 3]
```diff
@@ -41,41 +60,53 @@ export class ReceiptService {
...
+                        ReceiptType[receipt.type],
                         (receipt.recipientAddress instanceof Address ?
-                            receipt.recipientAddress.pretty() : receipt.recipientAddress.toHex()) + '\n';
-                    txt += '\t\tsenderPublickey:\t' + receipt.sender.publicKey + '\n';
-                    txt += '\t\tmosaicId:\t[ ' + receipt.mosaicId.id.lower + ', ' + receipt.mosaicId.id.higher + ' ]\n\n';
-                    txt += '\t\tamount:\t\t' + receipt.amount.compact() + '\n';
+                            receipt.recipientAddress.pretty() : receipt.recipientAddress.toHex()),
+                        receipt.sender.publicKey,
+                        '',
```
Comment (dgarcia360) [inline src/service/receipt.service.ts]: Why empty?

File: src/service/receipt.service.ts [hunk 4]
```diff
@@ -41,41 +60,53 @@ export class ReceiptService {
...
-                    txt += 'version:\t' + receipt.version + '\n';
-                    txt += '\t\ttype:\t\t' + ReceiptType[receipt.type] + '\n';
-                    txt += '\t\ttargetPublicKey:\t' + receipt.targetPublicAccount.publicKey + '\n';
-                    txt += '\t\tmosaicId:\t[ ' + receipt.mosaicId.id.lower + ', '
-                        + receipt.mosaicId.id.higher + ' ]\n\n';
-                    txt += '\t\tamount:\t\t' + receipt.amount.compact() + '\n';
+                    this.transactionStatementsTable.push([
+                        receipt.version as Table.CellOptions,
+                        ReceiptType[receipt.type],
+                        '',
```
Comment (dgarcia360) [inline src/service/receipt.service.ts]: Why empty?

File: src/service/receipt.service.ts [hunk 5]
```diff
@@ -41,41 +60,53 @@ export class ReceiptService {
     public formatTransactionStatements(statement: any): string {
         let txt = '';
         txt += chalk.green('transactionStatements:\t') + '\n';
-        txt += '-'.repeat('transactionStatements:\t'.length) + '\n\n';
         statement.transactionStatements.map((transaction: TransactionStatement, transactionIndex: number) => {
-            txt += 'height:\t\t' + transaction.height + '\n';
             transaction.receipts.map((receipt: any, receiptIndex: number) => {
-                txt += '<index: ' + transactionIndex + '-' + receiptIndex + '>\t';
                 if (receipt instanceof BalanceTransferReceipt) {
```
Comment (dgarcia360) [inline src/service/receipt.service.ts]: I'd create one class per receipt table, so we can test each table in the future.

Example: https://github.com/nemtech/nem2-cli/blob/master/src/commands/block/header.ts#L36

File: src/service/receipt.service.ts (hunk 1 above)
Comment (dgarcia360) [inline src/service/receipt.service.ts]: Maybe the transactionStatementsTable is not needed. We could print the header ``transactionStatements + source``, and then all the related receipts formatted as table individually.

File: src/service/receipt.service.ts (hunk 2 above)
Comment (NemTechCN) [reply src/service/receipt.service.ts]: The instance of `receipt .version` is `ReceiptType` which can not cast to CellOptions. So I cast it to Table.CellOptions Mandatorily. 
In fact, the value of `receipt.version` is 1 which is different from `ReceiptType`.

Comment (dgarcia360) [reply src/service/receipt.service.ts]: I believe you can cast version (integer) as a string (receipt.version.toString())

File: src/service/receipt.service.ts (hunk 1 above)
Comment (NemTechCN) [reply src/service/receipt.service.ts]: I don't understand this clearly. Could you please describe it more clearly?

**Comment by dgarcia360** [comment]
Thanks for your PR! Unfortunately, we consider it's better not to implement this change at the moment since it would be even more confusing for the users when rendering lots of receipts.

**Pull request closed without merging**